src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -307,6 +307,89 @@ def _is_range_full_extent(my_range):
     return is_lower and is_upper and is_step
 
 
+def default_precision():
```

Comment:
I've implemented the default_* functions which make things much nicer thanks. I've also added the type name as an argument to default_precision() so it can specify type-specific precision in the future.